components/provider_cldr/src/transform/timezones/cldr_json/mod.rs

```diff
@@ -0,0 +1,110 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/master/LICENSE ).
+mod convert;
+
+use crate::cldr_langid::CldrLangID;
+use serde::{Deserialize, Deserializer, de::{IgnoredAny, MapAccess, Visitor}};
+use std::collections::BTreeMap;
+
+#[derive(PartialEq, Debug, Clone, Deserialize)]
+pub struct MetaZone {
+    pub long: Option<ZoneFormat>,
+    pub short: Option<ZoneFormat>,
+}
+
+#[derive(PartialEq, Debug, Clone, Deserialize)]
+pub struct MetaZones(pub BTreeMap<String, MetaZone>);
+
+#[derive(PartialEq, Debug, Clone, Deserialize)]
+pub struct ZoneFormat(pub BTreeMap<String, String>);
+
+#[derive(PartialEq, Debug, Default, Clone)]
+pub struct TimeZoneNames {
```

Comment:
Optional: You can use `#[derive(serde::Deserialize)]` to extract the "normal" keys as usual, and then put a `#[serde(flatten)]` at the bottom with the custom `serde::de::Visitor` to process any leftover items in the map.

Reply:
I'm going to keep this as-is for now. 

I may explore with flatten in a separate PR.